File: RemixedDungeon/src/main/java/com/watabou/pixeldungeon/actors/buffs/Light.java

package com.watabou.pixeldungeon.actors.buffs;

import com.watabou.pixeldungeon.actors.Char;
import com.watabou.pixeldungeon.sprites.CharSprite;
import com.watabou.pixeldungeon.ui.BuffIndicator;

import org.jetbrains.annotations.NotNull;

public class Light extends FlavourBuff {

	public static final float DURATION	= 250f;
	
	@Override
	public boolean attachTo(@NotNull Char target ) {
		if (super.attachTo( target )) {
			target.observe();
			return true;
		} else {
			return false;
		}
	}
	
	@Override
	public void detach() {
		super.detach();
		target.observe();
	}
	
	@Override
	public int icon() {
		return BuffIndicator.LIGHT;
	}

	@Override
	public CharSprite.State charSpriteStatus() {
		return CharSprite.State.ILLUMINATED;
	}
}

File: scripts/buffs/Light.lua
---
--- Light: +1 view distance while active (was actors/buffs/Light.java)
---
local RPD  = require "scripts/lib/commonClasses"
local buff = require "scripts/lib/buff"

return buff.init{
    desc  = function ()
        return {
            icon          = 22, -- BuffIndicator.LIGHT
            name          = "LightBuff_Name",
            info          = "LightBuff_Info",
        }
    end,

    attached = function(self, buff)
        buff.target:observe()
    end,

    detach = function(self, buff)
        buff.target:observe()
    end,

    charSpriteStatus = function(self, buff)
        return "ILLUMINATED"
    end
}
